packages/cli/src/commands/git/connect.ts

```diff
@@ -101,10 +121,194 @@ export default async function connect(
+  const parsedRepoArg = parseRepoUrl(repoArg);
+  if (!parsedRepoArg) {
+    output.error(
+      `Cannot parse URL ${repoArg}. Make sure you didn't make any typos.`
+    );
+    return 1;
+  }
+  const { provider, org: parsedOrg, repo } = parsedRepoArg;
+  const repoPath = `${parsedOrg}/${repo}`;
+  const connect = await checkExistsAndConnect(
+    client,
+    output,
+    confirm,
+    org,
+    project,
+    project.link,
+    provider,
+    repoPath,
+    parsedOrg,
+    repo
+  );
+  if (typeof connect === 'number') {
+    return connect;
+  }
+  output.log(
+    `Connected ${formatProvider(provider)} repository ${chalk.cyan(repoPath)}!`
+  );
+  return 0;
+}
+
+async function connectArgWithLocalGit(
+  client: Client,
+  output: Output,
+  org: Org,
+  project: Project,
+  confirm: boolean,
+  gitConfig: Dictionary<any>,
+  repoArg: string
+) {
+  const parsedUrlArg = parseRepoUrl(repoArg);
+  if (!parsedUrlArg) {
+    output.error(
+      `Cannot parse URL ${repoArg}. Make sure you didn't make any typos.`
+    );
+    return 1;
+  }
+
+  const remoteUrls = pluckRemoteUrls(gitConfig);
+  if (remoteUrls) {
+    const shouldConnect = await promptConnectArg(
+      client,
+      output,
+      confirm,
+      repoArg,
+      remoteUrls
+    );
+    if (typeof shouldConnect === 'number') {
+      return shouldConnect;
+    }
+    if (shouldConnect) {
+      const { provider, org: parsedOrg, repo } = parsedUrlArg;
+      const repoPath = `${parsedOrg}/${repo}`;
+      output.log(`Connecting Git remote: ${link(repoArg)}`);
+      const connect = await checkExistsAndConnect(
+        client,
+        output,
+        confirm,
+        org,
+        project,
+        project.link,
+        provider,
+        repoPath,
+        parsedOrg,
+        repo
+      );
+      if (typeof connect === 'number') {
+        return connect;
+      }
+      output.log(
+        `Connected ${formatProvider(provider)} repository ${chalk.cyan(
+          repoPath
+        )}!`
+      );
+    }
+    return 0;
+  }
+  return await connectArg(client, output, confirm, org, project, repoArg);
+}
+
+async function promptConnectArg(
+  client: Client,
+  output: Output,
+  yes: boolean,
+  arg: string,
+  remoteUrls: Dictionary<string>
+) {
+  const multiple = Object.keys(remoteUrls).length > 1;
+  if (multiple) {
+    output.log(`Found multiple Git repositories in your local Git config:`);
+    printRemoteUrls(output, remoteUrls);
+  } else {
+    const url = Object.values(remoteUrls)[0];
+    const parsedUrl = parseRepoUrl(url);
+    if (!parsedUrl) {
+      output.error(
+        `Cannot parse URL ${url}. Make sure you didn't make any typos.`
+      );
+      return 1;
+    }
+    const parsedUrlArg = parseRepoUrl(arg);
+    if (!parsedUrlArg) {
+      output.error(
+        `Cannot parse URL ${arg}. Make sure you didn't make any typos.`
+      );
+      return 1;
+    }
+    if (JSON.stringify(parsedUrl) === JSON.stringify(parsedUrlArg)) {
+      return true;
```

Comment:
Is there a reason this is returning `true` instead of `0`? 

The errors above return `1` so I would assume success should return `0`

Reply:
Good catch, thanks

Reply:
Oh actually, it was returning true because `promptConnectUrl()` returns a boolean, so `true` means the code should continue, not exit successfully.

I updated it to only return a boolean and exit with code 1 if it returns false.